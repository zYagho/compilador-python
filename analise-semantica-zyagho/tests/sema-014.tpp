{Aviso: Função 'func' declarada, mas não utilizada}
{Aviso: Coerção implícita do valor atribuído para 'a', 'inteiro' para 'flutuante'}
{Erro: Função 'func' deveria retornar 'inteiro', mas retorna 'flutuante'}
{Aviso: Variável 'b' declarada e inicializada, mas não utilizada}
{Aviso: Chamada recursiva para 'principal'}
{Erro: Função 'principal' deveria retornar 'inteiro', mas retorna 'vazio'}

flutuante: a
inteiro: b

inteiro func()
  a := 10
  retorna(a)
fim

inteiro principal()
	b := 18
	principal()
fim
