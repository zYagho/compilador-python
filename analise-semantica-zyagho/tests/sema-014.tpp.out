WAR-SEM-FUNC-DECL-NOT-USED
WAR-SEM-IMP-COERC-OF-NUM
ERR-SEM-FUNC-RET-TYPE-ERROR
WAR-SEM-VAR-DECL-INIT-NOT-USED
WAR-SEM-CALL-REC-FUNC-MAIN
ERR-SEM-FUNC-RET-TYPE-ERROR
